### PR TITLE
Id set outgoing mappers - added support for the incident fields

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/32775

## Must have
- [x] Tests

### CODE REVIEW

File: demisto_sdk/commands/common/update_id_set.py [hunk 1]
```diff
@@ -710,7 +711,14 @@ def get_mapper_data(path):
     mapping = json_data.get('mapping', {})
     for key, value in mapping.items():
         incidents_types.add(key)
-        incidents_fields = incidents_fields.union(set(value.get('internalMapping').keys()))
+        if type_ == 'mapping-outgoing':
+            internal_mapping = value.get('internalMapping')  # get the mapping
+            incident_fields_set = {  # incident fields are in the simple key of each key
+                internal_mapping.get(key, {}).get('simple') for key in internal_mapping.keys()}
+            incidents_fields = incidents_fields.union(incident_fields_set)
+        else:  # pre 6.0 no type field in the mapper and default is an incoming mapper
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:719]: pre 6.0 there weren't any mappers.

File: demisto_sdk/commands/common/update_id_set.py [hunk 2]
```diff
@@ -710,7 +711,14 @@ def get_mapper_data(path):
     mapping = json_data.get('mapping', {})
     for key, value in mapping.items():
         incidents_types.add(key)
-        incidents_fields = incidents_fields.union(set(value.get('internalMapping').keys()))
+        if type_ == 'mapping-outgoing':
+            internal_mapping = value.get('internalMapping')  # get the mapping
+            incident_fields_set = {  # incident fields are in the simple key of each key
+                internal_mapping.get(key, {}).get('simple') for key in internal_mapping.keys()}
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:717]: can't we use the complex option as well?

**Comment by bakatzir** [comment]
@teizenman please review the changes in repo.py

File: demisto_sdk/commands/common/update_id_set.py [hunk 3]
```diff
@@ -710,7 +711,24 @@ def get_mapper_data(path):
...
+            # incident fields are in the simple key or in complex.root key of each key
+            for internal_mapping_key in internal_mapping.keys():
+                fields_mapper = internal_mapping.get(internal_mapping_key, {})
+                if isinstance(fields_mapper, dict):
+                    incident_field_simple = fields_mapper.get('simple')
+                    if incident_field_simple:
+                        incident_fields_set.add(incident_field_simple)
+                    else:
+                        incident_field_complex = fields_mapper.get('complex', {})
+                        if isinstance(incident_field_complex, dict) and 'root' in incident_field_complex:
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:726]: i think `isinstance(incident_field_complex, dict)` will always be True

**Review by yaakovi — APPROVED** [review]

**Pull request merged**